src/components/course/data/hooks.jsx

```diff
@@ -845,11 +845,11 @@ export const useExternalEnrollmentFailureReason = () => {
   ]);
 };
 
-export const useIsCourseAssigned = (redeemableLCPolicies, courseKey) => {
-  if (!redeemableLCPolicies || redeemableLCPolicies.length < 1) { return false; }
-
-  const learnerContentAssignmentsArray = redeemableLCPolicies.flatMap(item => item?.learnerContentAssignments || []);
-  if (!learnerContentAssignmentsArray) { return false; }
-
-  return learnerContentAssignmentsArray.some(assignment => assignment?.contentKey === courseKey);
+export const useIsCourseAssigned = (learnerContentAssignments, courseKey) => {
```

Comment:
[inform] Relies on the existing `learnerContentAssignments` instead of extracting out the assignments from `redeemableLCPolicies`.